README.md
Static File Cache (nc_staticfilecache)
======================================
**TYPO3 extension nc_staticfilecache**  

Transparent static file cache solution using mod_rewrite and mod_expires.  
Increase response times for static pages by a factor of up to 230!!!
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
Versions 1.0.0 - 3.6.0 have been completely provided by the `Static File Cache team`.   
These Versions have been downloaded here: 
 - https://extensions.typo3.org/extension/nc_staticfilecache/  
 - https://github.com/TYPO3-extensions/nc_staticfilecache  

Version 4.0.0 has been provided by Christopher Kaster:
 - https://github.com/atomicptr/nc_staticfilecache

The versions provided here are copies but versioned and tagged with git,
with this additional text-file `README.md` and **composer-support**.  

To install via composer use:  
```
composer require wdb/nc_staticfilecache
```

Versions
--------
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
4.0.0 / stable  
Aug 28, 2017	
 - Compatibility for TYPO3 version 8.7. Thank's to Christopher Kaster!
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
3.6.0 / stable  
March 23, 2016	
 - Now requires PHP 5.5 or higher, due to many code cleanups incl. PSR-2.  
 - [FEATURE] Add TSFE pageCacheTags to our own cache.  
 - Add nginx configuration to documentation.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
3.5.0 / stable  
December 28, 2015  
  Various bugfixes and code cleanups.  
  All currently known issues have been resolved.  
  Thanks to Tim Lochmueller for continued contributions!  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
3.4.1 / stable  
September 02, 2015  
**NOTE: .htaccess changes are needed when upgrading! Please refer to the manual!**  
 - Fix https://forge.typo3.org/issues/66842  
 - Remove the ValidProtocol Rule  
 - Add a security rule to prevent direct calls of static cache entries  
 - Remove the dummy htaccess file  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
3.3.1 / stable  
May 13, 2015  
 - Finished migration to CachingFramework.  
 - Compatibility until TYPO3 7.2.  
 - Thanks to Tim Lochmueller.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
3.2.0 / stable  
April 28, 2015  
 - The TYPO3 CachingFramework is used now.  
 - Cleanup of the Backend module.  
 - Documentation migrated to RestructuredText.  
 - Thanks to Tim Lochmueller for his extensive work.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
3.0.0 / stable  
April 16, 2015  
 - Vastly rewritten v3.0.0 for TYPO3 6.2.x - 7.1.x.  
 - Thanks to the contributors Tim Lochmüller and Jürgen Kußmann!  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
2.5.2 / stable  
November 11, 2014  
 - Bugfixes. With thanks to Tim Lochmüller.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
2.5.1 / stable  
April 09, 2014  
 - Fixed extension key  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
2.4.1 / stable  
March 11, 2014
 - Bugfix  
 - TYPO3 6.2 compatibility  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
2.4.0 / stable  
March 07, 2014  
 - Updated to svn v2.3.8 from forge.typo3.org  
 - Integrated TYPO3 6 compatibility adjustments for release as v2.4.0.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
2.3.4 / stable  
October 31, 2010  
 - See Changelog  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
2.3.1 / stable  
May 11, 2008  
 - fixed bug, $timeOutSeconds was set only when $conf[sendCacheControlHeader] was set but used also it $conf[sendCacheControlHeader] was not set . . . This caused expiry time to be 0 if sendCacheControlHeader was not set.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
2.3.0 / stable   
June 28, 2007  
 - Now handles external url properly, thanks to Helmut Hummel.  
 - Default exclusion of .css and .xml from cache rewriting.  
 - Put cleaner script back in. Its optional! Allows for static dynamic extensions.  
 - updated manual  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
2.2.0 / stable  
June 22, 2007  
 - Now supports simulateStaticDocuments, thanks to Marc Hörsken!  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
2.1.0 / stable  
June 14, 2007  
 - frontend user login detection with own cookie  
 - updated manual  
 - no more negative cache headers  
 - cache headers send on access just like with TYPO3 cache  
 - removal of cleaner script (not needed any longer because of new header code)  
 - added TODO to manual  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
2.0.0 / stable  
May 03, 2007  
 - Added info module to troubleshoot indexing.  
 - Some minor bugfixes.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  
1.0.0 / stable  
March 15, 2007  
 - Initial release.  
 - Speed up the serving of static pages using mod_rewrite and mod_expires.  
 - Speed increase factor of up to 92.  
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  


Changelog
2014-03-11  Klaus Bitto  <klaus@netcreators.nl>

	* Fixed bug: Call to undefined method TYPO3\CMS\Core\Utility\GeneralUtility::intval_positive() in class.tx_ncstaticfilecache_tasks_processDirtyPages_AdditionalFieldProvider.php (Thank you, Hendrik Reimers)
	* Increased compatibility to TYPO3 6.2.

2014-03-07  Klaus Bitto  <klaus@netcreators.nl>

	* Integrated TYPO3 6 compatibility adjustments for TER release as v2.4.0. (Selective merge from https://svn.typo3.org/TYPO3v4/Extensions/nc_staticfilecache/trunk/.)

2010-10-13  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Database elements are removed if directory on the filesystem exists, but could not be accessed

2010-09-20  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Removing of static files returned wrong boolean value
	* Fixed bug #9850: Small coding errors (thanks to Axel Jung)
	* Raised version to 2.3.3

2010-09-08  Oliver Hader  <oliver@typo3.org>

	* Cleanup: Fixed svn:eol-style of PHP and text files
	* Fixed bug: Only remove database elements if removal in filesystem was successful
	* Added feature: Integrate logging to devLog if clearing static caches failes
	* Raised version to 2.3.2

2010-07-19  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Typing error in hook name
	* Fixed bug: Infomodule shows creation time instead of last modification

2010-07-15  Oliver Hader  <oliver@typo3.org>

	* Cleanup: Fixed naming and formatting
	* Cleanup: Removed superfluous hook in processDirtyPages() method that was only available in Trunk

2010-07-14  Franz Ripfel  <franz.ripfel@abezet.de>

	* Fixed bug: Clearing cache of a single page deleted also all folders and files of subpages

2010-07-13  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: TYPO3 cache gets cleared on removing expired pages with the markDirtyInsteadOfDeletion setting enabled (thanks to Juergen Kussmann)

2010-05-28  Oliver Hader  <oliver@typo3.org>

	* Added feature: Integrate possibility to disable the clear cache post processing on deman during runtime

2010-05-27  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: markDirtyInsteadOfDeletion property shall only consider specific pages - thus not clear all or pages cache
	* Cleanup: Fixed formatting and inline type hints
	* Fixed bug: Database element is not removed if clearing files did not succeed
	* Fixed bug: Pages with an endtime that would expire a page before the general expiration time is not considered
	* Fixed bug: Additional hash is not written for database elements
	* Fixed bug: Additional hash is not considered for lookups when empty

2010-05-25  Oliver Hader  <oliver@typo3.org>

	* Added feature: Integrate hook to post process the cache scenario after (no matter whether static cache was written)

2010-04-30  Oliver Hader  <oliver@typo3.org>

	* Added feature: Integrate hook to handle deleting a static cached directory

2010-04-19  Oliver Hader  <oliver@typo3.org>

	* Follow-up to bug #5290: Expect the scheme name at first position and allow to modifiy with hook

2010-04-15  Oliver Hader  <oliver@typo3.org>

	* Follow-up to feature of predefining/extending values that are stored in the database
	* Added feature: Add additionalhash to implement individual and more specific database elements (utilized by hooks)
	* Fixed bug: removeExpiredPages triggeres clearing cache of a page multiple times

2010-04-14  Oliver Hader  <oliver@typo3.org>

	* Cleanup: Moved logging part of writing cache files to accordant place
	* Cleanup: Moved information that determine whether a page is cachable and added to them to hook parameters
	* Cleanup: Moved implementation to write compressed content to separate method
	* Cleanup: Extended parameters of createFile_processContent hook by URI and hostname
	* Cleanup: Renamed internal variable name
	* Added feature: Add possibility to predefine/extend values that are stored in the database

2010-02-22  Michiel Roos  <michiel@netcreators.com>

	* Updated the manual

2010-02-20  Michiel Roos  <michiel@netcreators.com>

	* Feature #3286: Enable usage of value 'reg1' from cache pages (Thanks to Alienor.net)

2010-02-19  Michiel Roos  <michiel@netcreators.com>

	* Feature #4179: Create gzipped versions of cache files (Thanks to Steffen Gebert)
	* Fixed bug #5290: nc_staticfilecache caches contents of https pages! (Thanks to Stefan Galinski)
	* Fixed bug #6525: EM refers to cc_devlog (Thanks to Steffen Gebert)

2010-02-17  Michiel Roos  <michiel@netcreators.com>

	* Fixed bug #6504: port based installations doesn't work (Thanks to Stefan Galinski)

2010-01-30  Michiel Roos  <michiel@netcreators.com>

	* Change: Show original URI on hover in infomodule

2010-01-22  Michiel Roos  <michiel@netcreators.com>

	* Fixed bug #6158: Scheduler tasks: missing ext_autoload (Thanks to Peter Schuster)

2010-01-14  Michiel Roos  <michiel@netcreators.com>

	* Fixed bug #4715: List what element are of INT type. (Thanks to Mads Jensen)
	* Added feature #6026: Provide scheduler tasks (Thanks to Michael Klapper)

2010-01-14  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: tx_ncstaticfilecache::processDirtyPages() removes entries from diry queue even if the processing did not succeed

2009-08-31  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Method tx_ncstaticfilecache::deleteStaticCacheDirectory() is protected but should be public

2009-08-13  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Current page Id is not outputted in form of backend info module

2009-08-10  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Visualization of tree in backend info module
	* Fixed bug: Visaulization does not depend on selected page of real page branch in backend info module
	* Fixed bug: Markup is incorrect on rendering the table in the backend info module
	* Fixed bug: Expanding/collapsing did not stay at the selected page in the backend info module

2009-08-07  Oliver Hader  <oliver@typo3.org>

	* Added feature: Integrate possibility to disable static caching for a page branch (tx_ncstaticfilecache.disableCache)


tasks/class.tx_ncstaticfilecache_tasks_processDirtyPages.php
<?php
/***************************************************************
 *  Copyright notice
 *
 *  (c) 2009 AOE media (dev@aoemedia.de)
 *  All rights reserved
 *
 *  This script is part of the TYPO3 project. The TYPO3 project is
 *  free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  The GNU General Public License can be found at
 *  http://www.gnu.org/copyleft/gpl.html.
 *
 *  This script is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  This copyright notice MUST APPEAR in all copies of the script!
 ***************************************************************/

require_once(t3lib_extMgm::extPath('nc_staticfilecache') . 'class.tx_ncstaticfilecache.php');

/**
 * {@inheritdoc}
 *
 * @author Michael Klapper <michael.klapper@aoemedia.de>
 * @copyright Copyright (c) 2009, AOE media GmbH <dev@aoemedia.de>
 * @version $Id: class.tx_ncstaticfilecache_tasks_processDirtyPages.php 76570 2013-06-14 18:28:11Z fab1en $
 * @date $Date$
 * @since 08.01.2010 - 11:00:44
 * @package TYPO3
 * @subpackage tx_ncstaticfilecache
 * @access public
 */
class tx_ncstaticfilecache_tasks_processDirtyPages extends tx_scheduler_Task {

	/**
	 * @var integer
	 */
	public $itemLimit = 0;


	/**
	 * This is the main method that is called when a task is executed
	 * It MUST be implemented by all classes inheriting from this one
	 * Note that there is no error handling, errors and failures are expected
	 * to be handled and logged by the client implementations.
	 * Should return true on successful execution, false on error.
	 *
	 * @access public
	 * @return boolean	Returns true on successful execution, false on error
	 *
	 * @author Michael Klapper <michael.klapper@aoemedia.de>
	 */
	public function execute() {
		/* @var $cleaner tx_ncstaticfilecache */
		$cleaner = t3lib_div::makeInstance('tx_ncstaticfilecache');
		$cleaner->processDirtyPages(NULL, $this->itemLimit);

		return true;
	}
}
?>


tasks/class.tx_ncstaticfilecache_tasks_removeExpiredPages.php
<?php
/***************************************************************
 *  Copyright notice
 *
 *  (c) 2009 AOE media (dev@aoemedia.de)
 *  All rights reserved
 *
 *  This script is part of the TYPO3 project. The TYPO3 project is
 *  free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  The GNU General Public License can be found at
 *  http://www.gnu.org/copyleft/gpl.html.
 *
 *  This script is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  This copyright notice MUST APPEAR in all copies of the script!
 ***************************************************************/

require_once(t3lib_extMgm::extPath('nc_staticfilecache') . 'class.tx_ncstaticfilecache.php');

/**
 * {@inheritdoc}
 *
 * @author Michael Klapper <michael.klapper@aoemedia.de>
 * @copyright Copyright (c) 2009, AOE media GmbH <dev@aoemedia.de>
 * @version $Id: class.tx_ncstaticfilecache_tasks_removeExpiredPages.php 76570 2013-06-14 18:28:11Z fab1en $
 * @date $Date$
 * @since 08.01.2010 - 11:00:44
 * @package TYPO3
 * @subpackage tx_ncstaticfilecache
 * @access public
 */
class tx_ncstaticfilecache_tasks_removeExpiredPages extends tx_scheduler_Task {

	/**
	 * This is the main method that is called when a task is executed
	 * It MUST be implemented by all classes inheriting from this one
	 * Note that there is no error handling, errors and failures are expected
	 * to be handled and logged by the client implementations.
	 * Should return true on successful execution, false on error.
	 *
	 * @access public
	 * @return boolean	Returns true on successful execution, false on error
	 *
	 * @author Michael Klapper <michael.klapper@aoemedia.de>
	 */
	public function execute() {
		/* @var $cleaner tx_ncstaticfilecache */
		$cleaner = t3lib_div::makeInstance('tx_ncstaticfilecache');
		$cleaner->removeExpiredPages();

		return true;
	}
}
?>

tasks/class.tx_ncstaticfilecache_tasks_processDirtyPages_AdditionalFieldProvider.php
<?php
/***************************************************************
 *  Copyright notice
 *
 *  (c) 2009 AOE media (dev@aoemedia.de)
 *  All rights reserved
 *
 *  This script is part of the TYPO3 project. The TYPO3 project is
 *  free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  The GNU General Public License can be found at
 *  http://www.gnu.org/copyleft/gpl.html.
 *
 *  This script is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  This copyright notice MUST APPEAR in all copies of the script!
 ***************************************************************/

/**
 * {@inheritdoc}
 *
 * @author Michael Klapper <michael.klapper@aoemedia.de>
 * @copyright Copyright (c) 2009, AOE media GmbH <dev@aoemedia.de>
 * @version $Id: class.tx_ncstaticfilecache_tasks_processDirtyPages_AdditionalFieldProvider.php 84493 2014-03-25 18:40:56Z spyker $
 * @date $Date$
 * @since 08.01.2010 - 11:00:44
 * @package TYPO3
 * @subpackage tx_ncstaticfilecache
 * @access public
 */
class tx_ncstaticfilecache_tasks_processDirtyPages_AdditionalFieldProvider implements tx_scheduler_AdditionalFieldProvider {

	/**
	 * This method is used to define new fields for adding or editing a task
	 * In this case, it adds an email field
	 *
	 * @param	array					$taskInfo: reference to the array containing the info used in the add/edit form
	 * @param	object					$task: when editing, reference to the current task object. Null when adding.
	 * @param	tx_scheduler_Module		$parentObject: reference to the calling object (Scheduler's BE module)
	 * @return	array					Array containg all the information pertaining to the additional fields
	 *									The array is multidimensional, keyed to the task class name and each field's id
	 *									For each field it provides an associative sub-array with the following:
	 *										['code']		=> The HTML code for the field
	 *										['label']		=> The label of the field (possibly localized)
	 *										['cshKey']		=> The CSH key for the field
	 *										['cshLabel']	=> The code of the CSH label
	 */
	public function getAdditionalFields(array &$taskInfo, $task, tx_scheduler_Module $schedulerModule) {
		$additionalFields = array();

		if (empty($taskInfo['itemLimit'])) {
                        if ($schedulerModule->CMD == 'add') {
                                $taskInfo['itemLimit'] = 0;
                        } else {
                                $taskInfo['itemLimit'] = $task->itemLimit;
                        }
		}
		$fieldID = 'task_itemLimit';
		$fieldCode  = '<input type="text" name="tx_scheduler[itemLimit]" id="' . $fieldID . '" value="' . $taskInfo['itemLimit'] . '" />';
		$additionalFields[$fieldID] = array(
                        'code'     => $fieldCode,
                        'label'    => 'LLL:EXT:nc_staticfilecache/locallang_db.xml:nc_staticfilecache_task_processDirtyPages.itemLimit'
                );
		return $additionalFields;
	}

	/**
	 * Validates the additional fields' values
	 *
	 * @param	array					An array containing the data submitted by the add/edit task form
	 * @param	tx_scheduler_Module		Reference to the scheduler backend module
	 * @return	boolean					True if validation was ok (or selected class is not relevant), false otherwise
	 */
	public function validateAdditionalFields(array &$submittedData, tx_scheduler_Module $schedulerModule) {

		$itemLimit = class_exists('t3lib_utility_Math') ?
			t3lib_utility_Math::convertToPositiveInteger($submittedData['itemLimit']) :
			t3lib_div::intval_positive($submittedData['itemLimit']);

		if ( $itemLimit > 0 ) {
			return true;
		} else {
			$schedulerModule->addMessage('no valid limit given (positive number expected)', t3lib_FlashMessage::ERROR);
			return false;
		}

	}

	/**
	 * Takes care of saving the additional fields' values in the task's object
	 *
	 * @param	array					An array containing the data submitted by the add/edit task form
	 * @param	tx_scheduler_Module		Reference to the scheduler backend module
	 * @return	void
	 */
	public function saveAdditionalFields(array $submittedData, tx_scheduler_Task $task) {
		$task->itemLimit = $submittedData['itemLimit'];
	}
}
?>


tasks/class.tx_ncstaticfilecache_tasks_removeExpiredPages_AdditionalFieldProvider.php
<?php
/***************************************************************
 *  Copyright notice
 *
 *  (c) 2009 AOE media (dev@aoemedia.de)
 *  All rights reserved
 *
 *  This script is part of the TYPO3 project. The TYPO3 project is
 *  free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  The GNU General Public License can be found at
 *  http://www.gnu.org/copyleft/gpl.html.
 *
 *  This script is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  This copyright notice MUST APPEAR in all copies of the script!
 ***************************************************************/

/**
 * {@inheritdoc}
 *
 * @author Michael Klapper <michael.klapper@aoemedia.de>
 * @copyright Copyright (c) 2009, AOE media GmbH <dev@aoemedia.de>
 * @version $Id: class.tx_ncstaticfilecache_tasks_removeExpiredPages_AdditionalFieldProvider.php 76570 2013-06-14 18:28:11Z fab1en $
 * @date $Date$
 * @since 08.01.2010 - 11:00:44
 * @package TYPO3
 * @subpackage tx_ncstaticfilecache
 * @access public
 */
class tx_ncstaticfilecache_tasks_removeExpiredPages_AdditionalFieldProvider {

	/**
	 * This method is used to define new fields for adding or editing a task
	 * In this case, it adds an email field
	 *
	 * @param	array					$taskInfo: reference to the array containing the info used in the add/edit form
	 * @param	object					$task: when editing, reference to the current task object. Null when adding.
	 * @param	tx_scheduler_Module		$parentObject: reference to the calling object (Scheduler's BE module)
	 * @return	array					Array containg all the information pertaining to the additional fields
	 *									The array is multidimensional, keyed to the task class name and each field's id
	 *									For each field it provides an associative sub-array with the following:
	 *										['code']		=> The HTML code for the field
	 *										['label']		=> The label of the field (possibly localized)
	 *										['cshKey']		=> The CSH key for the field
	 *										['cshLabel']	=> The code of the CSH label
	 */
	public function getAdditionalFields(array &$taskInfo, $task, tx_scheduler_Module $schedulerModule) {
		return array();
	}

	/**
	 * Validates the additional fields' values
	 *
	 * @param	array					An array containing the data submitted by the add/edit task form
	 * @param	tx_scheduler_Module		Reference to the scheduler backend module
	 * @return	boolean					True if validation was ok (or selected class is not relevant), false otherwise
	 */
	public function validateAdditionalFields(array &$submittedData, tx_scheduler_Module $schedulerModule) {
		return true;
	}

	/**
	 * Takes care of saving the additional fields' values in the task's object
	 *
	 * @param	array					An array containing the data submitted by the add/edit task form
	 * @param	tx_scheduler_Module		Reference to the scheduler backend module
	 * @return	void
	 */
	public function saveAdditionalFields(array $submittedData, tx_scheduler_Task $task) {
		return null;
	}
}
?>

ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "nc_staticfilecache".
 *
 * Auto generated 25-03-2014 07:22
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = [
    'title' => 'Static File Cache',
    'description' => 'Transparent static file cache solution using mod_rewrite and mod_expires. Increase response times for static pages by a factor of 230!!',
    'category' => 'fe',
    'version' => '4.0.0',
    'state' => 'beta',
    'modify_tables' => 'pages',
    'clearcacheonload' => true,
    'author' => 'Static File Cache team',
    'author_email' => 'extensions@netcreators.com',
    'author_company' => 'Netcreators',
    'constraints' => [
        'depends' => [
            'typo3' => '8.7.0-8.7.99',
            'php' => '7.0.0-0.0.0',
        ],
    ],
    'autoload' => [
        'psr-4' => [
            'SFC\\NcStaticfilecache\\' => 'Classes'
        ],
    ],
];
